crates/core/src/sql/kind.rs

```diff
@@ -15,37 +15,71 @@ use serde::{Deserialize, Serialize};
 use std::collections::BTreeMap;
 use std::fmt::{self, Display, Formatter, Write};
 
+/// The kind, or data type, of a value or field.
 #[revisioned(revision = 2)]
 #[derive(Clone, Debug, Eq, PartialEq, PartialOrd, Serialize, Deserialize, Hash)]
 #[cfg_attr(feature = "arbitrary", derive(arbitrary::Arbitrary))]
 #[non_exhaustive]
 pub enum Kind {
+	/// The most generic type, can be anything.
 	Any,
+	/// Null type.
 	Null,
+	/// Boolean type.
 	Bool,
+	/// Bytes type.
 	Bytes,
+	/// Datetime type.
 	Datetime,
+	/// Decimal type.
 	Decimal,
+	/// Duration type.
 	Duration,
+	/// 64-bit floating point type.
 	Float,
+	/// 64-bit signed integer type.
 	Int,
+	/// Number type, can be either a float, int or decimal.
+	/// This is the most generic type for numbers.
 	Number,
+	/// Object type.
 	Object,
+	/// Geometric 2D point type with x and y coordinates.
```

Comment:
Could specify here that it's longitude then latitude (GeoJSON spec), we have that in the docs in places but a user didn't notice yesterday so we added an even more obvious warning.